unittests/FrontendTool/FrontendObserverTests.cpp

```diff
@@ -30,20 +29,72 @@ class TestObserver : public testing::Test, public FrontendObserver {
 
 public:
 
-  void parsedArgs(CompilerInvocation &invocation) override { }
+  // convenient fields are grabbed from the given object at each method
+  // to assert that the passed object is valid and usable
+
+  // fields to assert
```

Comment:
Nitpick: Our comment style is formatted like an English sentence, including an initial capital letter and a final period.